<?xml version="1.0"?>
<launch>

  <arg name="online" default="false"/>
  <arg name="submap_frame" default="/kf_ref"/>
  <arg name="submap_topic" default="/lidar_keyframe_slam/kfRef_cloud"/>
  <arg name="slamMap_frame" default="/map"/>
  <arg name="slamMap_topic" default="/lidar_keyframe_slam/slamMap_cloud"/>
  <arg name="velodyne_frame" default="/velodyne"/>
  <arg name="velodyne_topic" default="/velodyne_points"/>
  
  <arg name="rviz" default="true"/>
  <arg name="rviz_config" default="/home/haowei/haowei_ws/src/semantic_labeller/config/lidar.rviz"/>

  <arg name="save_to_ply" default="false"/>
  <arg name="save_dir" default="/home/haowei/Documents/data/route_A/08-09-2019/frame2frame/crap9/"/>

  <!--full map pre-processing-->
  <arg name="min_intensity" default="50.0"/>
  <arg name="max_intensity" default="255.0"/>

  <!--radius to extract submap from full map-->
  <arg name="kd_search_radius" default="100.0"/>

  <!--normal estimation, either based on radius or number of nearest neighbors-->
  <arg name="normal_radius" default="0.3"/>
  <arg name="k_nearest" default="10"/>

  <!--ground plane estimation and extraction params-->
  <arg name="normal_ang_thresh" default="30.0"/>
  <arg name="RANSAC_iter" default="1000"/>
  <arg name="RANSAC_dist" default="0.1"/>
  <arg name="dist_to_plane" default="0.03"/>

  <!--dynamic and moving params-->
  <arg name="cons_scan_max_d" default="0.5"/>

  <!--dynamic not moving params-->
  <arg name="scan_to_map_max_d" default="0.5"/>

  <!-- always make sure we are not using simulation time when running online-->
  <param unless="$(arg online)" name="/use_sim_time" value="true"/>
  <param if="$(arg online)" name="/use_sim_time" value="false"/>

  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" output="screen" args=" -d $(arg rviz_config)"/>
  </group>

  <node pkg="semantic_labeller" type="semantic_labeller_node" name="semantic_labeller_node" output="screen">
    <param name="submap_frame" value="$(arg submap_frame)"/>
    <param name="submap_topic" value="$(arg submap_topic)"/>
    <param name="slamMap_frame" value="$(arg slamMap_frame)"/>
    <param name="slamMap_topic" value="$(arg slamMap_topic)"/>
    <param name="velodyne_frame" value="$(arg velodyne_frame)"/>
    <param name="velodyne_topic" value="$(arg velodyne_topic)"/>

    <param name="cons_scan_max_d" type="double" value="$(arg cons_scan_max_d)"/>
    <param name="scan_to_map_max_d" type="double" value="$(arg scan_to_map_max_d)"/>
    <param name="min_intensity" type="double" value="$(arg min_intensity)"/>
    <param name="max_intensity" type="double" value="$(arg max_intensity)"/>
    <param name="RANSAC_dist" type="double" value="$(arg RANSAC_dist)"/>
    <param name="RANSAC_iter" type="int" value="$(arg RANSAC_iter)"/>
    <param name="normal_radius" type="double" value="$(arg normal_radius)"/>
    <param name="normal_ang_thresh" type="double" value="$(arg normal_ang_thresh)"/>
    <param name="dist_to_plane" type="double" value="$(arg dist_to_plane)"/>
    <param name="k_nearest" type="int" value="$(arg k_nearest)"/>
    <param name="kd_search_radius" type="double" value="$(arg kd_search_radius)"/>

    <param name="save_to_ply" value="$(arg save_to_ply)"/>
    <param name="save_dir" value="$(arg save_dir)" />
  </node>

</launch>     
